File: NewsAlarm/app/src/main/java/com/alarm/newsalarm/newsmanager/JsonNewsParser.java
package com.alarm.newsalarm.newsmanager;

import androidx.annotation.NonNull;

import com.alarm.newsalarm.utils.LogUtil;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.HashSet;
import java.util.Set;

class JsonNewsParser {

    private static final String CLASS_NAME = "JsonNewsParser";

    Set<JSONObject> parseNaverNewsToJSONSet(String response) {
        try {
            JSONObject jsonArticles = new JSONObject(response);
            JSONArray jsonArticleArray = jsonArticles.getJSONArray("items");
            LogUtil.logD(CLASS_NAME, "parseNaverNewsToJSONSet", "json articles ready");

            return selectOnlyNaverNews(jsonArticleArray);
        } catch (JSONException e) {
            LogUtil.logE(CLASS_NAME, "parseNaverNewsToJSONSet", e);
        }
        return new HashSet<>();
    }

    @NonNull
    private Set<JSONObject> selectOnlyNaverNews(JSONArray jsonArticleArray) throws JSONException {
        Set<JSONObject> jsonArticleSet = new HashSet<>();
        for (int i = 0; i < jsonArticleArray.length(); i++) {
            JSONObject jsonArticle = jsonArticleArray.getJSONObject(i);
            if (jsonArticle.getString("link").contains("news.naver.com")) {
                jsonArticleSet.add(jsonArticle);
            }
            if (jsonArticleSet.size() == 5) {
                break;
            }
        }
        LogUtil.logI(CLASS_NAME, "selectOnlyNaverNews",
            "count of naver news article : " + jsonArticleSet.size());

        return jsonArticleSet;
    }
}

File: NewsAlarmKt/app/src/main/java/com/alarm/newsalarmkt/newsmanager/JsonNewsParser.kt
package com.alarm.newsalarmkt.newsmanager

import com.alarm.newsalarmkt.utils.LogUtil
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

class JsonNewsParser {

    fun parseNaverNewsToJSONSet(response: String): MutableSet<JSONObject> {
        try {
            val jsonArticles = JSONObject(response)
            val jsonArticleArray = jsonArticles.getJSONArray("items")
            LogUtil.logD(CLASS_NAME, "parseNaverNewsToJSONSet", "json articles ready")

            return selectOnlyNaverNews(jsonArticleArray)
        } catch (e: JSONException) {
            LogUtil.logE(CLASS_NAME, "parseNaverNewsToJSONSet", e)
        }
        return HashSet()
    }

    @Throws(JSONException::class)
    private fun selectOnlyNaverNews(jsonArticleArray: JSONArray): MutableSet<JSONObject> {
        val jsonArticleSet: MutableSet<JSONObject> = HashSet()
        for (i in 0..<jsonArticleArray.length()) {
            val jsonArticle = jsonArticleArray.getJSONObject(i)
            if (jsonArticle.getString("link").contains("news.naver.com")) {
                jsonArticleSet.add(jsonArticle)
            }
            if (jsonArticleSet.size == 5) {
                break
            }
        }
        LogUtil.logI(
            CLASS_NAME, "selectOnlyNaverNews",
            "count of naver news article : " + jsonArticleSet.size
        )

        return jsonArticleSet
    }

    companion object {
        private const val CLASS_NAME = "JsonNewsParser"
    }
}
